### PR TITLE
Fix more clippy warnings

### PR DESCRIPTION
**Pull request by liuw**

### CODE REVIEW

File: src/sexp_internal.rs [hunk 1]
```diff
@@ -13,10 +13,7 @@ pub fn display_cil(expr: &sexp::Sexp) -> String {
         Sexp::List(l) => {
             format!(
                 "({})",
-                l.iter()
-                    .map(display_cil)
-                    .collect::<Vec<String>>()
-                    .join(" ")
+                l.iter().map(display_cil).collect::<Vec<String>>().join(" ")
```
Comment (l0kod) [inline src/sexp_internal.rs:16]: Could you please squash this into the previous commit that changes the map call?

File: src/error.rs [hunk 2]
```diff
@@ -198,6 +198,12 @@ pub struct HLLErrors {
     errors: Vec<HLLErrorItem>,
 }
 
+impl Default for HLLErrors {
```
Comment (l0kod) [inline src/error.rs:201]: It seems weird to implement unused code just to please Clippy.

Comment (liuw) [reply src/error.rs:201]: I'm fine with dropping this patch. When clippy is enabled in the CI you will need to explicitly disable that lint.

**Review by dburgener — APPROVED** [review]

**Comment by l0kod** [comment]
Thanks!

**Pull request merged**